src/main/java/fridgy/ui/MainWindow.java

```diff
@@ -118,20 +118,40 @@ private void setAccelerator(MenuItem menuItem, KeyCombination keyCombination) {
         });
     }
 
+    /**
+     * Initialize all the stateful components
+     */
+    private void initialize() {
+        UiState uiState = new UiState(this);
+        logic.setUiState(uiState);
+
+        activeItemPanel = new ActiveItemPanel(logic::isEnough);
+        tabListPanel = new TabListPanel(
+            new IngredientListPanel(logic.getFilteredIngredientList(), uiState::setActive),
+            new RecipeListPanel(logic.getFilteredRecipeList(), uiState::setActive, logic::isEnough)
+        );
+
+
+        // Initialise event listeners.
+        this.getRoot().addEventFilter(ActiveItemChangeEvent.RECIPE, activeItemPanel::handleRecipeEvent);
+        this.getRoot().addEventFilter(ActiveItemChangeEvent.INGREDIENT, activeItemPanel::handleIngredientEvent);
+        this.getRoot().addEventFilter(ActiveItemChangeEvent.CLEAR, activeItemPanel::handleClearEvent);
+
+        this.getRoot().addEventFilter(TabSwitchEvent.CHANGE, tabListPanel::handleTabChange);
+        this.getRoot().addEventFilter(ActiveItemChangeEvent.RECIPE, tabListPanel::handleActiveRecipe);
+        this.getRoot().addEventFilter(ActiveItemChangeEvent.INGREDIENT, tabListPanel::handleActiveIngredient);
```

Comment:
Very good use of event handlers! 🐒

Reply:
Ok, changing delete command to switch tab and clear window if deleted item is active.